engine/util/src/main/scala/pl/touk/nussknacker/engine/util/exception/WithResources.scala

```diff
@@ -0,0 +1,30 @@
+package pl.touk.nussknacker.engine.util.exception
+
+object WithResources {
+
+  // based on Oracle try-with-resources explanation from
+  // https://www.oracle.com/technical-resources/articles/java/trywithresources.html
+  def use[Resource <: AutoCloseable, Result](resource: Resource)
+                                            (consumer: Resource => Result): Result = {
+
+    var maybeThrown: Throwable = null
```

Comment:
Please try do this `try in try`.. as recursion function, after that you should be able to remove `var maybeThrown: Throwable = null`.